udark/udapp/src/main/java/io/underdark/app/model/Node.java

```diff
@@ -119,9 +123,10 @@ public void sendFrame(byte[] frameData, Link link){
 
 	//protocolId: 0 = send routingtable, 1 = del destination, 2 = random message
 	public void sendFrame(Link link, int protocolId, String str){
-		String newStr = Integer.toString(protocolId) + str;
+		//protocol id | sender id | reciever id | payload
+		String newStr = Integer.toString(protocolId) + "|" + link.getNodeId() + "|" + this.nodeId + "|" + str;
```

Comment:
We should only attach this metadata if we are not in protocol 0 or 1. When we exchange routing information, we will only ever communicate with people within our naturally discovered links. No need for this extra information/parsing of it on the receiving end when in modes 0 or 1.

Reply:
nvm was tired when i made this comment....misread

Reply:
lol wait no I'm right we only append the meta info when not in mode 0 or 1

Reply:
Also, if you always put in link.getNodeId, you lose the original sender of the message after 1 hop